sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedDeltaBinaryPackedReader.java

```diff
@@ -300,8 +300,12 @@ private void unpackMiniBlock() throws IOException {
         bitWidths[currentMiniBlock]);
     for (int j = 0; j < miniBlockSizeInValues; j += 8) {
       ByteBuffer buffer = in.slice(packer.getBitWidth());
-      packer.unpack8Values(buffer.array(),
-        buffer.arrayOffset() + buffer.position(), unpackedValuesBuffer, j);
+      if (buffer.hasArray()) {
+        packer.unpack8Values(buffer.array(),
+          buffer.arrayOffset() + buffer.position(), unpackedValuesBuffer, j);
+      } else {
+        packer.unpack8Values(buffer, buffer.position(), unpackedValuesBuffer, j);
```

Comment:
nit:  should we add a tips or todo due to UT can't cover this branch now, I think we should test it in the future

Reply:
How do you mean? Do you mean that there is no unit test to check? I can add a test that covers both cases.

Reply:
I mean line 307  should not be covered by the UT now

Reply:
Yes, there are no unit tests for this. Even before this patch, this would only be covered if the encoding is enabled and data pages v2 are used.